com.microsoft.mrtk.accessibility/Subsystems/MRTK/MRTKAccessibilitySubsystem.cs

```diff
@@ -40,10 +41,148 @@ class AccessibilityProvider : Provider
+                List<GameObject> objectList)
+            {
+                if ((int)(classifications & ObjectClassification.People) != 0)
+                {
+                    objectList.AddRange(describableObjects[ObjectClassification.People]);
+                }
+                if ((int)(classifications & ObjectClassification.Places) != 0)
+                {
+                    objectList.AddRange(describableObjects[ObjectClassification.Places]);
+                }
+                if ((int)(classifications & ObjectClassification.Things) != 0)
+                {
+                    objectList.AddRange(describableObjects[ObjectClassification.Things]);
+                }
+                if ((int)(classifications & ObjectClassification.UserInterface) != 0)
+                {
+                    objectList.AddRange(describableObjects[ObjectClassification.UserInterface]);
+                }
+            }
+
+            /// <summary>
+            /// Filters the provided collection of <see cref="GameObject"/>s based on the visibility to the main camera
+            /// and the maximum allowed distance.
+            /// </summary>
+            /// <param name="readerView">The desired visibility (in field of view, full surround, etc.) of the <see cref="GameObject"/>s.</param>
+            /// <param name="maxDistance">The distance, in meters, beyond which <see cref="GameObject"/>s will be filtered.</param>
+            /// <param name="objectList">The collection of <see cref="GameObject"/>s that will be filtered based on visibility and distance.</param>
+            private void FilterDescribableObjects(
+                ReaderView readerView,
+                float maxDistance,
+                List<GameObject> objectList)
+            {
+                // Ensure there is a collection to filter.
+                if (objectList.Count == 0) { return; }
+
+                // Walk the list backwards, so that it does not hit a collection changed exception.
+                for (int i = (objectList.Count - 1); i != 0; i--)
+                {
+                    GameObject obj = objectList[i];
+
+                    // Does the object have a collider?
+                    Collider collider = obj.GetComponent<Collider>();
+                    if (collider == null)
+                    {
+                        objectList.Remove(obj);
+                        continue;
+                    }
+
+                    // If requested, is it within the main camera's field of view?
+                    if ((readerView == ReaderView.FieldOfView) &&
+                        !CameraFOVChecker.IsInFOVCached(Camera.main, collider))
+                    {
+                        objectList.Remove(obj);
+                    }
+
+                    // Is it closer than the maximum distance?
+                    if (maxDistance < Mathf.Abs(Vector3.Distance(Camera.main.transform.position, obj.transform.position)))
```

Comment:
A potential perf improvement:

`Vector3.Distance` is the equivalent of `(Camera.main.transform.position - obj.transform.position).magnitude`, which includes a square root calculation that can be expensive on some architectures.

Outside of the `for` loop, you could cache `maxDistance` squared, and then compare that against `(Camera.main.transform.position - obj.transform.position).sqrMagnitude` to save on some operations.

Reply:
done. thanks!